---
title: "CVX Molecular Selection"
author: "Boris Igic"
date: "5/27/2021"
output: html_document
---
Note:rerun by AT on 6/13/22
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
```

## Plotting Molecular Selection Results from DataMonkey

This is code for plotting the output of DataMonkey. CVX has five annotated genes in our data, so Boris took the individuals files and ran them through Fast Unconstrained Bayesian AppRoximation (FUBAR).

## FUBAR

*FUBAR (Fast, Unconstrained Bayesian AppRoximation)* uses a Bayesian approach to infer nonsynoymous (dN, here labeled *beta*) and synonymous (dS, *alpha*) substitution rates on a per-site basis for a given coding alignment and corresponding phylogeny. This method assumes that the selection pressure for each site is constant along the entire phylogeny.

Although FUBAR produces similar information to FEL, it has several key differences. FUBAR employs a Bayesian algorithm to infer rates, and therefore it reports evidence for positive selection using posterior probabilities (which range from 0-1), not p-values. Generally, posterior probabilities > 0.9 are strongly suggestive of positive selection.

FUBAR runs extremely quickly and is well-suited for analyzing large alignments, with hundreds or thousands of sequences. This speed-up results from the novel strategy of employing a pre-specified discrete grid of dN and dS values to be applied across sites. This approach contrasts with the time-consuming FEL strategy of fitting a new MG94xREV model at each site. *FUBAR may have more power than FEL, in particular when positive selection is present but relatively weak (i.e. low values of ω>1).*

If you use FUBAR in your analysis, please cite the following: Murrell, B et al. "FUBAR: A Fast, Unconstrained Bayesian AppRoximation for inferring selection." Mol. Biol. Evol. 30, 1196–1205 (2013).  doi:10.1093/molbev/mst030.

## CVX 

The genes are CP, TGB1-3, and RdRp. The input files are named <gene order number>_ + <gene-name>_ + "nostop.fasta"
e.g., <5_CP_nostop.fasta>. 

TGB3 & CP reading frames are overlapping, so the analyses here include a replicated bit of sequence, but in different reading frames.

### Import and Prepare Data

```{r data-prep}
# RdRp<-read.csv("/Users/boris/Dropbox/PROJECTS/Alexa/cactusvirusx/analyses/CVX-selection/FUBAR/RdRp_60961fdc238adf71a515f4a8/datamonkey-table.csv")
# TGB1<-read.csv("/Users/boris/Dropbox/PROJECTS/Alexa/cactusvirusx/analyses/CVX-selection/FUBAR/TGB1_6096ed5d238adf71a5160d3b/datamonkey-table.csv")
# TGB2<-read.csv("/Users/boris/Dropbox/PROJECTS/Alexa/cactusvirusx/analyses/CVX-selection/FUBAR/TGB2_6096f188238adf71a5160e3e/datamonkey-table.csv")
# TGB3<-read.csv("/Users/boris/Dropbox/PROJECTS/Alexa/cactusvirusx/analyses/CVX-selection/FUBAR/TGB3_6096f1c9238adf71a5160e9e/datamonkey-table.csv")
# CP<-read.csv("/Users/boris/Dropbox/PROJECTS/Alexa/cactusvirusx/analyses/CVX-selection/FUBAR/CP_6096240f238adf71a515f5a1/datamonkey-table.csv")
RdRp<-read.csv("RdRp_60961fdc238adf71a515f4a8/datamonkey-table.csv")
TGB1<-read.csv("TGB1_6096ed5d238adf71a5160d3b/datamonkey-table.csv")
TGB2<-read.csv("TGB2_6096f188238adf71a5160e3e/datamonkey-table.csv")
TGB3<-read.csv("TGB3_6096f1c9238adf71a5160e9e/datamonkey-table.csv")
CP<-read.csv("CP_6096240f238adf71a515f5a1/datamonkey-table.csv")
# The names()
# 
# "Site"
# "Partition"
# "X.beta."
# "X.beta...alpha."
# "Prob..alpha...beta.."
# "Prob..alpha...beta...1"
# "BayesFactor..alpha...beta.."
# [will add new site number below]

bar.w = function(i=7){
  barplot(c(RdRp[,i],TGB1[,i],TGB2[,i],TGB3[,i],CP[,i]))
}
bar.w()

alpha=c(RdRp[,3],TGB1[,3],TGB2[,3],TGB3[,3],CP[,3])
beta=c(RdRp[,4],TGB1[,4],TGB2[,4],TGB3[,4],CP[,4])
Site=1:length(beta)
cvx.sel<-data.frame(Site, alpha, beta)

allsites<-rbind(RdRp,TGB1,TGB2,TGB3,CP)

# Now Site gives the original site for EACH GENE (replicates early numbers)
#
# totSite refers to site in whole CVX (total)
allsites$totSite<-1:length(allsites$Site)

names(allsites)<-c("Site","Partition","alpha","beta","difference","prob.a_gt_b","prob.a_lt_b","BayesFactor.a_lt_b","totSite")
```

## Plots

```{r}
# Sliding window dN/dS
ggplot(allsites, aes(totSite,zoo::rollmean(beta, 10, na.pad=T))) + geom_line() + theme_bw() + labs( x = "Site", y = "Beta")

# Probability dN > dS
ggplot(allsites, aes(totSite,zoo::rollmean(prob.a_lt_b, 1, na.pad=T))) + geom_line() + theme_light() + labs( x = "Site", y = "prob.a_lt_b")

# Sliding window dN/dS
# Note that this is for replicated regions of overlap!
# (length is greater than total length of alignment)
ggplot(allsites, aes(totSite,zoo::rollmean(beta/alpha, 10, na.pad=T))) + 
  geom_line() + 
  geom_line(aes(totSite,zoo::rollmean(beta, 10, na.pad=T)), color = "red", linetype = "dotted") +
  geom_line(aes(totSite,zoo::rollmean(alpha, 10, na.pad=T)), color = "blue", linetype = "dotted") + 
# Adding gene position segments
  # RdRp
  geom_segment(aes(x = 0, y = -0.1, xend = dim(RdRp)[1], yend = -0.1),color = "gray") +
  # TGB1
  geom_segment(aes(x = dim(RdRp)[1], y = -0.2, xend = sum(dim(RdRp)[1],dim(TGB1)[1]), yend = -0.2), color = "gray") +
  # TGB2
  geom_segment(aes(x = sum(dim(RdRp)[1],dim(TGB1)[1]), y = -0.1, xend = sum(dim(RdRp)[1],dim(TGB1)[1],dim(TGB2)[1]), yend = -0.1), color = "grey") +
  # TGB3
  geom_segment(aes(x = sum(dim(RdRp)[1],dim(TGB1)[1],dim(TGB2)[1]), y = -0.2, xend = sum(dim(RdRp)[1],dim(TGB1)[1],dim(TGB2)[1],dim(TGB3)[1]), yend = -0.2), color = "grey") +
  # CP
  geom_segment(aes(x = sum(dim(RdRp)[1],dim(TGB1)[1],dim(TGB2)[1],dim(TGB3)[1]), y = -0.1, xend = sum(dim(RdRp)[1],dim(TGB1)[1],dim(TGB2)[1],dim(TGB3)[1],dim(CP)[1]), yend = -0.1), color = "grey") +
  #15kd protein average start and end calculated based on geneious annotations
  geom_segment(aes(x = 471.3, y = -0.2, xend = 605, yend = -0.2), color = "grey") +
  #8kd protein
    geom_segment(aes(x = 974.47, y = -0.2, xend = 1047, yend = -0.2), color = "grey") +
   ##extra ORFS here:
   # geom_segment(aes(x = START_SITE, y = -0.1, xend = END_SITE, yend = -0.1), color = "grey") +
  
  theme_light() + 
  labs( x = "Site", y = "dN/dS")
ggsave(filename = "selectionplot.pdf", width = 10, height = 5)

# BY EACH GENE sliding window dN/dS

ggplot(RdRp, aes(Site,zoo::rollmean(RdRp[,4]/RdRp[,3], 10, na.pad=T))) + geom_line() + theme_light() + labs( x = "Site", y = "dN/dS")

ggplot(TGB1, aes(Site,zoo::rollmean(TGB1[,4]/TGB1[,3], 10, na.pad=T))) + geom_line() + theme_light() + labs( x = "Site", y = "dN/dS")

ggplot(TGB2, aes(Site,zoo::rollmean(TGB2[,4]/TGB2[,3], 10, na.pad=T))) + geom_line() + theme_light() + labs( x = "Site", y = "dN/dS")

ggplot(TGB3, aes(Site,zoo::rollmean(TGB3[,4]/TGB3[,3], 10, na.pad=T))) + geom_line() + theme_light() + labs( x = "Site", y = "")

ggplot(CP, aes(Site,zoo::rollmean(CP[,4]/CP[,3], 10, na.pad=T))) + geom_line() + theme_light() + labs( x = "Site", y = "")
```



## To do: 

- Draw horizontal lines with dN/DS averages for each gene, which will also label the individual genes and indicate areas of overlapping reading frames
