---
title: "cvx_phylogeny_annotation"
author: "Alexa & Boris"
date: date: "Last compiled on `r format(Sys.time(), '%d %B, %Y')`"
output: html_document
---
To do:
-collapse branches
-recolor based on host/virus
-rename tips

> This is a one-time package installation to help get libraries likely to be missing at the start.

```{r eval=FALSE}
#install needed packages
install.packages('BiocManager')# This option gave B errors; repos=c('https://cran.rstudio.com'), ask=F)
BiocManager::install('DECIPHER')
BiocManager::install("ggtree")
BiocManager::install("phytools")
```

## Loading Libraries

```{r load-libraries}
library(ggtree) # tree plotting
library(DECIPHER)
library(RCurl) # web page loading
library(phangorn)
library(phytools)
```

## Data Files and Locations
>Note: Errors on windows still to be resolved

Specify location of RNA sequence alignment and metadata:
```{r}
metadata.loc <- 'https://raw.githubusercontent.com/pseudocrater/cactusvirusx/master/data/meatdata_v2.csv'
aln.fasta.loc <- 'https://raw.githubusercontent.com/pseudocrater/cactusvirusx/master/data/genbank_KR_viruses_aln_v2.fasta'
#Notes: These have been exported from geneious, aligned using MAFFT[KR], exported as both .fasta and as .csv for relevant columns.
#Note: For .fasta files, U has been converted to T, and then "KT8" to "KU8" to conserve naming.
#Note: hosts have been added to organisms that lacked them.

```

Fetch alignment and metadata, convert alignment:
```{r}
aln.fasta <- getURL(aln.fasta.loc)
seq.metadata.csv <-getURL(metadata.loc)
write(x=seq.metadata.csv, file='seq.metadata.csv')
```

Write (save) to working directory, then read again; beware of overwriting previous files!:

```{r}
cvx.metadata <- read.csv('seq.metadata.csv', stringsAsFactors = FALSE)
#adjust metadata for correct organism names

write(aln.fasta, file = 'aln.fasta')
cacti.aln <- read.dna(file = 'aln.fasta', format= "fasta")
aln.phy <- read.phyDat('aln.fasta', format='fasta', type='DNA')
aln.bin <- as.DNAbin(aln.phy)
```

## Making and Editing a Phylogenetic Tree

For now, make a simple guide neighbor-joining tree, which will be used for annotation.

```{r}
cacti.nj <- NJ(dist.dna(aln.bin))
```

> Note: it would likely be better to perform this on the alignment file, so that the alignment and tree file can be easily compared later, without having to back-convert name
s. Also, it is very useful to provide a detailed accounting of what "adjust" means. It's not adjusting, but renaming, and the rationale for the renaming is missing, as well as the sources of the introduced information. e.g.: why is "NC_002815" converted to "Cactus virus X_AF308158" and where is that information? This would be very, very, hard to understand or recover independently and the goal is to make a document for someone else to follow (or yourself much later).


Edit host names to add placeholder for unknown host, and root the tree:
>Note: figure out which 

```{r}
#Notes on translating ref seq names to organism names:  gsub("NC_002815", "Cactus virus X_AF308158"; "NC_006059", "Zygocactus virus X_AY366208"; "NC_011659", "Schlumbergera virus X_AY366207";"NC_024458", "Pitaya virus X_JF930327"
for (i in 1:length(host.names)) {
  if (host.names[i]==""){
    host.names[i] <- "Host unknown"
  }
}
#names <- cacti.nj$tip.label
#names <- append(names, rep("nodetest",89))
#hosts <- as.character(append(host.names, rep("nodetest",89)))
#df <- data.frame(names, hosts)
cacti.njtree <- cacti.nj
cacti.njtree <- root(cacti.nj, outgroup=c(88:90), resolve.root=T)

is.rooted(cacti.njtree)
```


```{r}
#plot.phylo(ladderize(cacti.njtree), type='phylo', no.margin=F, cex=0.5, label.offset=0.0025)
write.tree(cacti.njtree, file="cacti.njtree.nwk")
nj.gg <- read.tree(file="cacti.njtree.nwk")
#for (i in 1:length(nj.gg$tip.label)) {
 # nj.gg$tip.label[i] <- gsub("_", " ", x = nj.gg$tip.label[i])
  #nj.gg$tip.label[i] <- gsub("-", ",", x = nj.gg$tip.label[i])
  
#}
hosts.df <- data.frame(hosts)
is.rooted(nj.gg)
```

Rooting options.
```{r}
#ape:
#nj.gg <- unroot(nj.gg)
#nj.gg <- root(nj.gg, outgroup=c(88:90), resolve.root=FALSE)
#ape option 2:
#nj.gg <- ape::plot.phylo(nj.gg,  root.edge = T)
#phytools:
nj.gg<-phytools::reroot(nj.gg, node=91, edgelabel=TRUE, position=0)

#ggtree: (not working)
#nj.gg <-rroot(nj.gg, node=91, outgroup= c(88:90), resolve.root=TRUE)
```

Create tree using ggtree. 
```{r}
par( mar=c(1, 1, 0, 1))
ggtree(nj.gg, ladderize=T, aes(), show.legend = TRUE)+
#%<+% hosts.df %<+% df  +
  
  #name tip labels:
  geom_tiplab(aes(), hjust=0, offset=0, align=T, linetype = "dashed" )+
  #host tip labels:
#  geom_tiplab(aes(label=hosts.df$hosts), align = T, linetype = 'blank',
#              na.rm=FALSE, offset=0.2) +
  #scale
  geom_treescale(x=0.05, y=25, offset=2, fontsize = 3)+
  #numbered tip labels for checking data addition
  #geom_tiplab(aes(label=c(1:179)), size= 2, align = T, linetype = 'blank',
     #         na.rm=FALSE, offset=0.2) +
  #display/check node labels:
  #geom_text2(aes(subset=!isTip, label=node), hjust=-.3)+
  #highlight specified node:
  #geom_hilight(node=101,fill="yellow", alpha=.6)+
  xlim(0,0.75)
```

Save the tree to working directory in pdf format.
```{r}
filename = paste(gsub(":", "-", Sys.time()),"_names_hosts_nj.pdf",sep="")
ggsave(filename, width = 60, height = 40, units = "cm", limitsize = FALSE)
```

Parsimony calculations.
```{r}
#parsimony for nj tree
cacti.nj <- NJ(dist.dna(cacti.bin))
parsimony(tree=cacti.nj, data=aln.phy)
cacti.nj.optim <- optim.parsimony(cacti.nj, aln.phy)
cacti.nj.pratchet <- pratchet(aln.phy)
```

Cleaning up:
```{r}
unlink("aln.fasta")
unlink("seq.metadata.csv")
```

