---
title: "cvx_phylogeny_annotation"
author: "Alexa & Boris"
date: "Last compiled on `r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document: default
---

> This is a one-time package installation to help get libraries likely to be missing at the start.

```{r eval=FALSE}
knitr::opts_chunk$set(error = TRUE)
#install needed packages
install.packages('BiocManager')# This option gave B errors; repos=c('https://cran.rstudio.com'), ask=F)
BiocManager::install('DECIPHER')
BiocManager::install("ggtree")
BiocManager::install("phytools")
tinytex::install_tinytex()
install.packages('BiocManager', repos=c('https://cran.rstudio.com'), ask=FALSE)
BiocManager::install('DECIPHER')



```

## Loading Libraries

```{r load-libraries, message=FALSE}
library(ggtree) # tree plotting
library(DECIPHER)
library(RCurl) # web page loading
library(phangorn)
library(phytools)
library(ape)
library("knitr")
library(DECIPHER)
```

## Data Files and Locations
>Note: Errors on windows still to be resolved

Specify location of RNA sequence alignment and metadata:
```{r}
metadata.loc <- 'https://raw.githubusercontent.com/pseudocrater/cactusvirusx/master/data/cvx_hostdata_v3.csv'
cvx.unaln.fasta.loc <- 'https://raw.githubusercontent.com/pseudocrater/cactusvirusx/master/data/genbank_kr_consensus_extraction_unal.fasta'
aln.fasta.loc <- 'https://raw.githubusercontent.com/pseudocrater/cactusvirusx/master/data/genbank_KR_viruses_consensus_aln.fasta'

#Notes: These have been exported from geneious, aligned using MAFFT[KR], exported as both .fasta and as .csv for relevant columns.
#Note: For .fasta files, U has been converted to T, and then "KT8" to "KU8" to conserve naming. 
#KT15 ->KU15
#KT6 -> KU6
#Notes on translating ref seq names to organism names:  gsub("NC_002815", "Cactus virus X_AF308158"; "NC_006059", "Zygocactus virus X_AY366208"; "NC_011659", "Schlumbergera virus X_AY366207";"NC_024458", "Pitaya virus X_JF930327"
#Note: hosts have been added to organisms that lacked them.

```

Fetch alignment and metadata, convert alignment:
```{r}

cvx.init.aln.fasta <- getURL(aln.fasta.loc)
cvx.unaln.fasta <- getURL(cvx.unaln.fasta.loc)
write(cvx.unaln.fasta, file = 'cvx.unaln.fasta')



cvx.seq.metadata.csv <-getURL(metadata.loc)

write(x=cvx.seq.metadata.csv, file='cvx.seq.metadata.csv')

```

Alignment tools with DECIPHER. 

```{r}
#note: fasta file without gaps has been produced by geneious 11.5 extraction of sequences and deletion of the consensus sequence 

unal.f <- 'cvx.unaln.fasta'
unal.dna <- readDNAStringSet(unal.f)
#note: error appears, "reading FASTA file cvx.unaln.fasta: ignored 361 invalid one-letter sequence codes"
al.dan <- AlignSeqs(unal.dna)
BrowseSeqs(al.dan, highlight=1)
unlink('cvx.unaln.fasta')
```




Write (save) to working directory, then read again; beware of overwriting previous files!:

```{r}
cvx.metadata <- read.csv('cvx.seq.metadata.csv', stringsAsFactors = FALSE)
#alphabetize metadata, only needs to be done once:
#cvx.metadata <- cvx.metadata[order(cvx.metadata$Name),]
#write.csv(cvx.metadata, file="test.csv")
#cvx.metadata <- read.csv('test.csv', stringsAsFactors = FALSE)
write(x=cvx.seq.metadata.csv, file='cvx.seq.metadata.csv')


write(cvx.init.aln.fasta, file = 'cvx.init.aln.fasta')
cacti.aln <- read.dna(file = 'cvx.init.aln.fasta', format= "fasta")
cvx.aln.phy <- read.phyDat('cvx.init.aln.fasta', format='fasta', type='DNA')
cvx.aln.bin <- as.DNAbin(cvx.aln.phy)
```

## Making and Editing a Phylogenetic Tree

For now, make a simple guide neighbor-joining tree, which will be used for annotation.

```{r}
cvx.nj <- NJ(dist.dna(cvx.aln.bin))
```


```{r}
#cacti.nj <- root(cacti.nj, outgroup=c(7), resolve.root=T)
is.rooted(cvx.nj)
```


Rooting options.
```{r}
write.tree(cvx.nj, file="cvx.njtree.nwk")
nj.gg <- read.tree(file="cvx.njtree.nwk")
nj.gg <- midpoint.root(nj.gg)
```


Plotting tree using plot(), and choosing nodes:
```{r extract-clade, eval=FALSE, message=FALSE}
choose.node <- function(col,tree) {
    o <- identify(tree)
    nodelabels(node=o$nodes, pch = 19, col = col)
    print(o)
}

#plot(nj.gg, type="f",cex=0.5)

# Not run
# Selecting node subtending potexviruses
#choose.node("red",nj.gg)
#Click close to a node of the tree...
#$nodes
#[1] 379

# Selecting node subtending potexviruses
#choose.node("green",cvx.tre)
#Click close to a node of the tree...
#$nodes
#[1] 382

```

```{r tip-rename}
#create vector for host information
hostinfo <- data.frame(nj.gg$tip.label)
#rename tree tips and populate vector
tip.rename.org = function(tree,dataframe){
   for(i in 1:length(tree$tip.label)){
      hostinfo$nj.gg.tip.label[i] <<- dataframe$host[grep(tree$tip.label[i],dataframe$Name)]
      #uncomment these for troubleshooting:
      print(i)
      print(tree$tip.label[i])
      print(dataframe$Organism[grep(tree$tip.label[i],dataframe$Name)])
      tree$tip.label[i]<-dataframe$Organism[grep(tree$tip.label[i],dataframe$Name)]
        
   }
tree
}

nj.gg <- read.tree(file="cvx.njtree.nwk")
nj.gg <- midpoint.root(nj.gg)
nj.gg <- tip.rename.org(nj.gg,cvx.metadata)

```

Collapse nodes:
```{r}
#find internal node numbers:
#MRCA(nj.gg, 7, 9)


```

Create tree using ggtree. 
```{r fig.width=6}
par( mar=c(1, 1, 0, 0))
L <- nj.gg$Nnode + length(nj.gg$tip.label)
ggtree(nj.gg, ladderize=T, aes(), show.legend = TRUE, size=0.2)  %<+% hostinfo+
  #name tip labels:
  geom_tiplab(aes(label=nj.gg$tip.label[1:L]), hjust=0, offset=0, align=T, linetype = "dotted")+
  #host tip labels:
  geom_tiplab(aes(label=hostinfo$nj.gg.tip.label[1:L]), align = T, linetype = 'blank', na.rm=TRUE, offset=0.15) +
  #scale
  geom_treescale(x=0, y=60, offset=2, fontsize = 3)+
  #node labels:
  #geom_text(aes(label=node), hjust=0.2) +
  #geom_hilight(node=136, fill="steelblue", alpha=.6, extend=0.5)+
  xlim(0,0.6) 

```


Save the tree to folder in pdf format, if wanted:
```{r eval=F}
filename = paste(gsub(":", "-", Sys.time()),"_names_hosts_nj.pdf",sep="")
ggsave(filename, width = 60, height = 40, units = "cm", limitsize = FALSE)
```

Distance calculations, requiring list of the comparisons requested.
```{r}
#create example list, or use your own for specific pairs. for all combos use dist.nodes(nj.gg)
data <- c("Virus from SRR11190799 Plant: BSJ-3", "Virus from SRR11190791 Plant: BSJ-1", "Virus from SRR11190800 Plant: BSJ-2")
data <- c("Virus from SRR11190798 Plant: DH-3", "Virus from SRR11190801 Plant: DH-2")
data <- c("Virus from SRR11603183 GW-peel-2", "Virus from SRR11603182 GW-peel-3")
data <- c("Virus from 15H-03_pol_S48", "Virus from 15H3_sty_S73","Virus from 15H3__s_S50")
cross.data <- expand.grid(tip1 = data, tip2 = data)
cross.data <- expand.grid(tip1 = nj.gg$tip.label, tip2 = nj.gg$tip.label)

d<- c()
get.tree.dist = function(solcross.data,pathd8.tre)
{	
	for(i in 1:length(solcross.data$tip1))
	{
		ifelse(is.na(match(solcross.data$tip1[i],pathd8.tre$tip.label)) | is.na(match(solcross.data$tip2[i],pathd8.tre$tip.label)),d[i]<-NA,
		d[i]<-cophenetic.phylo(pathd8.tre)[solcross.data$tip1[i],solcross.data$tip2[i]])
	}
	return(d)
}
test <- get.tree.dist(cross.data,nj.gg)
test
mean(test[test!=0])
par( mar=c(2, 2, 2, 2))
heatmap(cophenetic.phylo(nj.gg), cexRow = 0.2, cexCol = 0.2)
```

Parsimony calculations:
```{r eval=F}
#parsimony for nj tree
cvx.nj <- NJ(dist.dna(cvx.bin))
parsimony(tree=cvx.nj, data=cvx.aln.phy)
cvx.nj.optim <- optim.parsimony(cvx.nj, cvx.aln.phy)
cvx.nj.pratchet <- pratchet(cvx.aln.phy)
```

Cleaning up:
```{r}
unlink("cvx.init.aln.fasta")
unlink("cvx.seq.metadata.csv")
unlink("cvx.njtree.nwk")
unlink("cvx.unaln.fasta")
```

