% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={cvx\_phylogeny\_annotation},
  pdfauthor={Alexa \& Boris},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering

\title{cvx\_phylogeny\_annotation}
\author{Alexa \& Boris}
\date{Last compiled on 03 March, 2021}

\begin{document}
\maketitle

To do: -add more hosts -collapse branches -recolor based on host/virus
-rename tips

\begin{quote}
This is a one-time package installation to help get libraries likely to
be missing at the start.
\end{quote}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\OperatorTok{::}\NormalTok{opts_chunk}\OperatorTok{$}\KeywordTok{set}\NormalTok{(}\DataTypeTok{error =} \OtherTok{TRUE}\NormalTok{)}
\CommentTok{#install needed packages}
\KeywordTok{install.packages}\NormalTok{(}\StringTok{'BiocManager'}\NormalTok{)}\CommentTok{# This option gave B errors; repos=c('https://cran.rstudio.com'), ask=F)}
\NormalTok{BiocManager}\OperatorTok{::}\KeywordTok{install}\NormalTok{(}\StringTok{'DECIPHER'}\NormalTok{)}
\NormalTok{BiocManager}\OperatorTok{::}\KeywordTok{install}\NormalTok{(}\StringTok{"ggtree"}\NormalTok{)}
\NormalTok{BiocManager}\OperatorTok{::}\KeywordTok{install}\NormalTok{(}\StringTok{"phytools"}\NormalTok{)}
\NormalTok{tinytex}\OperatorTok{::}\KeywordTok{install_tinytex}\NormalTok{()}
\NormalTok{rmarkdown}\OperatorTok{::}\KeywordTok{render}\NormalTok{(}\StringTok{'test.rmd'}\NormalTok{, }\DataTypeTok{output_format =} \StringTok{'html_document'}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{loading-libraries}{%
\subsection{Loading Libraries}\label{loading-libraries}}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(ggtree) }\CommentTok{# tree plotting}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Registered S3 method overwritten by 'treeio':
##   method     from
##   root.phylo ape
\end{verbatim}

\begin{verbatim}
## ggtree v2.4.1  For help: https://yulab-smu.top/treedata-book/
## 
## If you use ggtree in published research, please cite the most appropriate paper(s):
## 
## [36m-[39m Guangchuang Yu. Using ggtree to visualize data on tree-like structures. Current Protocols in Bioinformatics, 2020, 69:e96. doi:10.1002/cpbi.96
## [36m-[39m Guangchuang Yu, Tommy Tsan-Yuk Lam, Huachen Zhu, Yi Guan. Two methods for mapping and visualizing associated data on phylogeny using ggtree. Molecular Biology and Evolution 2018, 35(12):3041-3043. doi:10.1093/molbev/msy194
## [36m-[39m Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution 2017, 8(1):28-36. doi:10.1111/2041-210X.12628
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(DECIPHER)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: Biostrings
\end{verbatim}

\begin{verbatim}
## Loading required package: BiocGenerics
\end{verbatim}

\begin{verbatim}
## Loading required package: parallel
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'BiocGenerics'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min
\end{verbatim}

\begin{verbatim}
## Loading required package: S4Vectors
\end{verbatim}

\begin{verbatim}
## Loading required package: stats4
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'S4Vectors'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:ggtree':
## 
##     expand
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:base':
## 
##     expand.grid
\end{verbatim}

\begin{verbatim}
## Loading required package: IRanges
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'IRanges'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:ggtree':
## 
##     collapse
\end{verbatim}

\begin{verbatim}
## Loading required package: XVector
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'Biostrings'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:base':
## 
##     strsplit
\end{verbatim}

\begin{verbatim}
## Loading required package: RSQLite
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(RCurl) }\CommentTok{# web page loading}
\KeywordTok{library}\NormalTok{(phangorn)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: ape
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'ape'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:Biostrings':
## 
##     complement
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:ggtree':
## 
##     rotate
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(phytools)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: maps
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(ape)}
\KeywordTok{library}\NormalTok{(}\StringTok{"knitr"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{data-files-and-locations}{%
\subsection{Data Files and Locations}\label{data-files-and-locations}}

\begin{quote}
Note: Errors on windows still to be resolved
\end{quote}

Specify location of RNA sequence alignment and metadata:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{metadata.loc <-}\StringTok{ 'https://raw.githubusercontent.com/pseudocrater/cactusvirusx/master/data/meatdata_v3.csv'}
\NormalTok{aln.fasta.loc <-}\StringTok{ 'https://raw.githubusercontent.com/pseudocrater/cactusvirusx/master/data/genbank_KR_viruses_aln_v3.fasta'}
\CommentTok{#Notes: These have been exported from geneious, aligned using MAFFT[KR], exported as both .fasta and as .csv for relevant columns.}
\CommentTok{#Note: For .fasta files, U has been converted to T, and then "KT8" to "KU8" to conserve naming. }
\CommentTok{#KT15 ->KU15}
\CommentTok{#KT6 -> KU6}
\CommentTok{#Notes on translating ref seq names to organism names:  gsub("NC_002815", "Cactus virus X_AF308158"; "NC_006059", "Zygocactus virus X_AY366208"; "NC_011659", "Schlumbergera virus X_AY366207";"NC_024458", "Pitaya virus X_JF930327"}
\CommentTok{#Note: hosts have been added to organisms that lacked them.}
\end{Highlighting}
\end{Shaded}

Fetch alignment and metadata, convert alignment:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{aln.fasta <-}\StringTok{ }\KeywordTok{getURL}\NormalTok{(aln.fasta.loc)}
\NormalTok{seq.metadata.csv <-}\KeywordTok{getURL}\NormalTok{(metadata.loc)}
\KeywordTok{write}\NormalTok{(}\DataTypeTok{x=}\NormalTok{seq.metadata.csv, }\DataTypeTok{file=}\StringTok{'seq.metadata.csv'}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Write (save) to working directory, then read again; beware of
overwriting previous files!:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{cvx.metadata <-}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{'seq.metadata.csv'}\NormalTok{, }\DataTypeTok{stringsAsFactors =} \OtherTok{FALSE}\NormalTok{)}
\CommentTok{#alphabetize metadata, only needs to be done once:}
\CommentTok{#cvx.metadata <- cvx.metadata[order(cvx.metadata$Name),]}
\CommentTok{#write.csv(cvx.metadata, file="test.csv")}
\CommentTok{#cvx.metadata <- read.csv('test.csv', stringsAsFactors = FALSE)}
\KeywordTok{write}\NormalTok{(}\DataTypeTok{x=}\NormalTok{seq.metadata.csv, }\DataTypeTok{file=}\StringTok{'seq.metadata.csv'}\NormalTok{)}


\KeywordTok{write}\NormalTok{(aln.fasta, }\DataTypeTok{file =} \StringTok{'aln.fasta'}\NormalTok{)}
\NormalTok{cacti.aln <-}\StringTok{ }\KeywordTok{read.dna}\NormalTok{(}\DataTypeTok{file =} \StringTok{'aln.fasta'}\NormalTok{, }\DataTypeTok{format=} \StringTok{"fasta"}\NormalTok{)}
\NormalTok{aln.phy <-}\StringTok{ }\KeywordTok{read.phyDat}\NormalTok{(}\StringTok{'aln.fasta'}\NormalTok{, }\DataTypeTok{format=}\StringTok{'fasta'}\NormalTok{, }\DataTypeTok{type=}\StringTok{'DNA'}\NormalTok{)}
\NormalTok{aln.bin <-}\StringTok{ }\KeywordTok{as.DNAbin}\NormalTok{(aln.phy)}
\end{Highlighting}
\end{Shaded}

\hypertarget{making-and-editing-a-phylogenetic-tree}{%
\subsection{Making and Editing a Phylogenetic
Tree}\label{making-and-editing-a-phylogenetic-tree}}

For now, make a simple guide neighbor-joining tree, which will be used
for annotation.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{cacti.nj <-}\StringTok{ }\KeywordTok{NJ}\NormalTok{(}\KeywordTok{dist.dna}\NormalTok{(aln.bin))}
\end{Highlighting}
\end{Shaded}

\begin{quote}
Note: figure out which organisms and hosts belong to which viruses.
Then, root the tree appropriately with clover virus.
\end{quote}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#cacti.nj <- root(cacti.nj, outgroup=c(7), resolve.root=T)}
\KeywordTok{is.rooted}\NormalTok{(cacti.nj)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] FALSE
\end{verbatim}

Rooting options.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#plot.phylo(ladderize(cacti.njtree), type='phylo', no.margin=F, cex=0.5, label.offset=0.0025)}
\KeywordTok{write.tree}\NormalTok{(cacti.nj, }\DataTypeTok{file=}\StringTok{"cacti.njtree.nwk"}\NormalTok{)}
\NormalTok{nj.gg <-}\StringTok{ }\KeywordTok{read.tree}\NormalTok{(}\DataTypeTok{file=}\StringTok{"cacti.njtree.nwk"}\NormalTok{)}
\CommentTok{#for (i in 1:length(nj.gg$tip.label)) \{}
 \CommentTok{# nj.gg$tip.label[i] <- gsub("_", " ", x = nj.gg$tip.label[i])}
  \CommentTok{#nj.gg$tip.label[i] <- gsub("-", ",", x = nj.gg$tip.label[i])}
  
\CommentTok{#\}}
\KeywordTok{is.rooted}\NormalTok{(nj.gg)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] FALSE
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#find internal node number:}
\KeywordTok{MRCA}\NormalTok{(nj.gg, }\StringTok{"D29630.1"}\NormalTok{, }\StringTok{"GQ179647"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 138
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#ape:}
\NormalTok{nj.gg <-}\StringTok{ }\KeywordTok{unroot}\NormalTok{(nj.gg)}
\CommentTok{#phytools:}
\NormalTok{nj.gg<-phytools}\OperatorTok{::}\KeywordTok{reroot}\NormalTok{(nj.gg, }\DataTypeTok{node=}\DecValTok{138}\NormalTok{, }\DataTypeTok{edgelabel=}\OtherTok{TRUE}\NormalTok{, }\DataTypeTok{position=}\DecValTok{0}\NormalTok{)}

\CommentTok{#ggtree: (not working)}
\CommentTok{#nj.gg <-rroot(nj.gg, outgroup= c(88:90), resolve.root=TRUE)}
\end{Highlighting}
\end{Shaded}

Plotting tree using plot()

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{choose.node <-}\StringTok{ }\ControlFlowTok{function}\NormalTok{(col,tree) \{}
\NormalTok{    o <-}\StringTok{ }\KeywordTok{identify}\NormalTok{(tree)}
    \KeywordTok{nodelabels}\NormalTok{(}\DataTypeTok{node=}\NormalTok{o}\OperatorTok{$}\NormalTok{nodes, }\DataTypeTok{pch =} \DecValTok{19}\NormalTok{, }\DataTypeTok{col =}\NormalTok{ col)}
    \KeywordTok{print}\NormalTok{(o)}
\NormalTok{\}}

\KeywordTok{plot}\NormalTok{(nj.gg, }\DataTypeTok{type=}\StringTok{"f"}\NormalTok{,}\DataTypeTok{cex=}\FloatTok{0.5}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{cvx_phylogeny_annotation_files/figure-latex/extract-clade-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# Not run}
\CommentTok{# Selecting node subtending potexviruses}
\CommentTok{#choose.node("red",nj.gg)}
\CommentTok{#Click close to a node of the tree...}
\CommentTok{#$nodes}
\CommentTok{#[1] 379}

\CommentTok{# Selecting node subtending potexviruses}
\CommentTok{#choose.node("green",cvx.tre)}
\CommentTok{#Click close to a node of the tree...}
\CommentTok{#$nodes}
\CommentTok{#[1] 382}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{#create vector for host information}
\NormalTok{hostinfo <-}\StringTok{ }\KeywordTok{data.frame}\NormalTok{(nj.gg}\OperatorTok{$}\NormalTok{tip.label)}
\CommentTok{#rename tree tips and populate vector}
\NormalTok{tip.rename.org =}\StringTok{ }\ControlFlowTok{function}\NormalTok{(tree,dataframe)\{}
   \ControlFlowTok{for}\NormalTok{(i }\ControlFlowTok{in} \DecValTok{1}\OperatorTok{:}\KeywordTok{length}\NormalTok{(tree}\OperatorTok{$}\NormalTok{tip.label))\{}
\NormalTok{      hostinfo[i] <<-}\StringTok{ }\NormalTok{dataframe}\OperatorTok{$}\NormalTok{host[}\KeywordTok{grep}\NormalTok{(tree}\OperatorTok{$}\NormalTok{tip.label[i],dataframe}\OperatorTok{$}\NormalTok{Name)]}
\NormalTok{      tree}\OperatorTok{$}\NormalTok{tip.label[i]<-dataframe}\OperatorTok{$}\NormalTok{Organism[}\KeywordTok{grep}\NormalTok{(tree}\OperatorTok{$}\NormalTok{tip.label[i],dataframe}\OperatorTok{$}\NormalTok{Name)]}
\NormalTok{   \}}
\NormalTok{hostinfo}
\NormalTok{tree}
\NormalTok{\}}

\NormalTok{nj.gg <-}\StringTok{ }\KeywordTok{read.tree}\NormalTok{(}\DataTypeTok{file=}\StringTok{"cacti.njtree.nwk"}\NormalTok{)}
\NormalTok{nj.gg <-}\StringTok{ }\KeywordTok{midpoint.root}\NormalTok{(nj.gg)}
\NormalTok{nj.gg <-}\StringTok{ }\KeywordTok{tip.rename.org}\NormalTok{(nj.gg,cvx.metadata)}
\end{Highlighting}
\end{Shaded}

Create tree using ggtree.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{par}\NormalTok{( }\DataTypeTok{mar=}\KeywordTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\DecValTok{4}\NormalTok{, }\DecValTok{0}\NormalTok{, }\DecValTok{0}\NormalTok{))}
\KeywordTok{ggtree}\NormalTok{(nj.gg, }\DataTypeTok{ladderize=}\NormalTok{T, }\KeywordTok{aes}\NormalTok{(), }\DataTypeTok{show.legend =} \OtherTok{TRUE}\NormalTok{,)  }\OperatorTok{%<+%}\StringTok{ }\NormalTok{hostinfo }\OperatorTok{+}
\StringTok{  }\CommentTok{#name tip labels:}
\StringTok{  }\KeywordTok{geom_tiplab}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{label=}\NormalTok{nj.gg}\OperatorTok{$}\NormalTok{tip.label[}\DecValTok{1}\OperatorTok{:}\DecValTok{211}\NormalTok{]), }\DataTypeTok{hjust=}\DecValTok{0}\NormalTok{, }\DataTypeTok{offset=}\DecValTok{0}\NormalTok{, }\DataTypeTok{align=}\NormalTok{T, }\DataTypeTok{linetype =} \StringTok{"dotted"}\NormalTok{)}\OperatorTok{+}
\StringTok{  }\CommentTok{#host tip labels:}
\StringTok{  }\KeywordTok{geom_tiplab}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{label=}\NormalTok{hostinfo}\OperatorTok{$}\NormalTok{nj.gg.tip.label[}\DecValTok{1}\OperatorTok{:}\DecValTok{211}\NormalTok{]), }\DataTypeTok{align =}\NormalTok{ T, }\DataTypeTok{linetype =} \StringTok{'blank'}\NormalTok{, }\DataTypeTok{na.rm=}\OtherTok{TRUE}\NormalTok{, }\DataTypeTok{offset=}\FloatTok{0.1}\NormalTok{) }\OperatorTok{+}
\StringTok{   }\CommentTok{# geom_tiplab(aes(label=cvx.metadata$Name[1:211]), align = T, linetype = 'blank', na.rm=TRUE, offset=0.2) +}
\StringTok{    }\CommentTok{#geom_tiplab(aes(label=nj.gg$tip.label[1:211]), align = T, linetype = 'blank', na.rm=TRUE, offset=0.3) +}
\StringTok{  }\CommentTok{#scale}
\StringTok{  }\KeywordTok{geom_treescale}\NormalTok{(}\DataTypeTok{x=}\DecValTok{0}\NormalTok{, }\DataTypeTok{y=}\DecValTok{60}\NormalTok{, }\DataTypeTok{offset=}\DecValTok{2}\NormalTok{, }\DataTypeTok{fontsize =} \DecValTok{3}\NormalTok{)}\OperatorTok{+}
\StringTok{  }\CommentTok{#node labels:}
\StringTok{  }\CommentTok{#geom_text(aes(label=), hjust=0.1) +}
\StringTok{  }\KeywordTok{xlim}\NormalTok{(}\DecValTok{0}\NormalTok{,}\FloatTok{1.2}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{cvx_phylogeny_annotation_files/figure-latex/unnamed-chunk-8-1.pdf}

Save the tree to folder in pdf format.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{filename =}\StringTok{ }\KeywordTok{paste}\NormalTok{(}\KeywordTok{gsub}\NormalTok{(}\StringTok{":"}\NormalTok{, }\StringTok{"-"}\NormalTok{, }\KeywordTok{Sys.time}\NormalTok{()),}\StringTok{"_names_hosts_nj.pdf"}\NormalTok{,}\DataTypeTok{sep=}\StringTok{""}\NormalTok{)}
\KeywordTok{ggsave}\NormalTok{(filename, }\DataTypeTok{width =} \DecValTok{60}\NormalTok{, }\DataTypeTok{height =} \DecValTok{40}\NormalTok{, }\DataTypeTok{units =} \StringTok{"cm"}\NormalTok{, }\DataTypeTok{limitsize =} \OtherTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Parsimony calculations.

Cleaning up:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{unlink}\NormalTok{(}\StringTok{"aln.fasta"}\NormalTok{)}
\KeywordTok{unlink}\NormalTok{(}\StringTok{"seq.metadata.csv"}\NormalTok{)}
\KeywordTok{unlink}\NormalTok{(}\StringTok{"cacti.njtree.nwk"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\end{document}
