---
title: "cvx_phylogeny_annotation"
author: "Alexa & Boris"
date: "Last compiled on `r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document: default
  html_notebook: default
  html_document: default
---
To do:
-add more hosts
-collapse branches
-recolor based on host/virus
-rename tips

> This is a one-time package installation to help get libraries likely to be missing at the start.

```{r eval=FALSE}
knitr::opts_chunk$set(error = TRUE)
#install needed packages
install.packages('BiocManager')# This option gave B errors; repos=c('https://cran.rstudio.com'), ask=F)
BiocManager::install('DECIPHER')
BiocManager::install("ggtree")
BiocManager::install("phytools")
tinytex::install_tinytex()
rmarkdown::render('test.rmd', output_format = 'html_document')

```

## Loading Libraries

```{r load-libraries}
library(ggtree) # tree plotting
library(DECIPHER)
library(RCurl) # web page loading
library(phangorn)
library(phytools)
library(ape)
library("knitr")
```

## Data Files and Locations
>Note: Errors on windows still to be resolved

Specify location of RNA sequence alignment and metadata:
```{r}
metadata.loc <- 'https://raw.githubusercontent.com/pseudocrater/cactusvirusx/master/data/meatdata_v3.csv'
aln.fasta.loc <- 'https://raw.githubusercontent.com/pseudocrater/cactusvirusx/master/data/genbank_KR_viruses_aln_v3.fasta'
#Notes: These have been exported from geneious, aligned using MAFFT[KR], exported as both .fasta and as .csv for relevant columns.
#Note: For .fasta files, U has been converted to T, and then "KT8" to "KU8" to conserve naming. 
#KT15 ->KU15
#KT6 -> KU6
#Notes on translating ref seq names to organism names:  gsub("NC_002815", "Cactus virus X_AF308158"; "NC_006059", "Zygocactus virus X_AY366208"; "NC_011659", "Schlumbergera virus X_AY366207";"NC_024458", "Pitaya virus X_JF930327"
#Note: hosts have been added to organisms that lacked them.

```

Fetch alignment and metadata, convert alignment:
```{r}
aln.fasta <- getURL(aln.fasta.loc)
seq.metadata.csv <-getURL(metadata.loc)
write(x=seq.metadata.csv, file='seq.metadata.csv')
```

Write (save) to working directory, then read again; beware of overwriting previous files!:

```{r}
cvx.metadata <- read.csv('seq.metadata.csv', stringsAsFactors = FALSE)
#alphabetize metadata, only needs to be done once:
#cvx.metadata <- cvx.metadata[order(cvx.metadata$Name),]
#write.csv(cvx.metadata, file="test.csv")
#cvx.metadata <- read.csv('test.csv', stringsAsFactors = FALSE)
write(x=seq.metadata.csv, file='seq.metadata.csv')


write(aln.fasta, file = 'aln.fasta')
cacti.aln <- read.dna(file = 'aln.fasta', format= "fasta")
aln.phy <- read.phyDat('aln.fasta', format='fasta', type='DNA')
aln.bin <- as.DNAbin(aln.phy)
```

## Making and Editing a Phylogenetic Tree

For now, make a simple guide neighbor-joining tree, which will be used for annotation.

```{r}
cacti.nj <- NJ(dist.dna(aln.bin))
```


>Note: figure out which organisms and hosts belong to which viruses. Then, root the tree appropriately with clover virus.

```{r}
#cacti.nj <- root(cacti.nj, outgroup=c(7), resolve.root=T)
is.rooted(cacti.nj)
```


Rooting options.
```{r}
#plot.phylo(ladderize(cacti.njtree), type='phylo', no.margin=F, cex=0.5, label.offset=0.0025)
write.tree(cacti.nj, file="cacti.njtree.nwk")
nj.gg <- read.tree(file="cacti.njtree.nwk")
#for (i in 1:length(nj.gg$tip.label)) {
 # nj.gg$tip.label[i] <- gsub("_", " ", x = nj.gg$tip.label[i])
  #nj.gg$tip.label[i] <- gsub("-", ",", x = nj.gg$tip.label[i])
  
#}
is.rooted(nj.gg)
#find internal node number:
MRCA(nj.gg, "D29630.1", "GQ179647")
#ape:
nj.gg <- unroot(nj.gg)
#phytools:
nj.gg<-phytools::reroot(nj.gg, node=138, edgelabel=TRUE, position=0)

#ggtree: (not working)
#nj.gg <-rroot(nj.gg, outgroup= c(88:90), resolve.root=TRUE)
```
Plotting tree using plot()
```{r extract-clade}
choose.node <- function(col,tree) {
    o <- identify(tree)
    nodelabels(node=o$nodes, pch = 19, col = col)
    print(o)
}

plot(nj.gg, type="f",cex=0.5)

# Not run
# Selecting node subtending potexviruses
choose.node("red",nj.gg)
#Click close to a node of the tree...
#$nodes
#[1] 379

# Selecting node subtending potexviruses
#choose.node("green",cvx.tre)
#Click close to a node of the tree...
#$nodes
#[1] 382

```

```{r tip-rename}
#create vector for host information
hostinfo <- data.frame(nj.gg$tip.label)
#rename tree tips and populate vector
tip.rename.org = function(tree,dataframe){
   for(i in 1:length(tree$tip.label)){
      hostinfo[i] <<- dataframe$host[grep(tree$tip.label[i],dataframe$Name)]
      tree$tip.label[i]<-dataframe$Organism[grep(tree$tip.label[i],dataframe$Name)]
   }
hostinfo
tree
}

nj.gg <- read.tree(file="cacti.njtree.nwk")
nj.gg <- midpoint.root(nj.gg)
nj.gg.v1 <- tip.rename.org(nj.gg,cvx.metadata)

```



Create tree using ggtree. 
```{r}
par( mar=c(1, 4, 0, 0))
ggtree(nj.gg.v1, ladderize=T, aes(), show.legend = TRUE,)  %<+% hostinfo +
  #name tip labels:
  geom_tiplab(aes(label=nj.gg.v1$tip.label[1:210]), hjust=0, offset=0, align=T, linetype = "dotted")+
  #host tip labels:
  geom_tiplab(aes(label=hostinfo$[1:210]), align = T, linetype = 'blank', na.rm=TRUE, offset=0.1) +
   # geom_tiplab(aes(label=cvx.metadata$Name[1:211]), align = T, linetype = 'blank', na.rm=TRUE, offset=0.2) +
    #geom_tiplab(aes(label=nj.gg$tip.label[1:211]), align = T, linetype = 'blank', na.rm=TRUE, offset=0.3) +
  #scale
  geom_treescale(x=0, y=60, offset=2, fontsize = 3)+
  #node labels:
  #geom_text(aes(label=), hjust=0.1) +
  xlim(0,1.2)
```

Save the tree to folder in pdf format.
```{r}
filename = paste(gsub(":", "-", Sys.time()),"_names_hosts_nj.pdf",sep="")
ggsave(filename, width = 60, height = 40, units = "cm", limitsize = FALSE)
```

Parsimony calculations.
```{r eval=F, echo=F}
#parsimony for nj tree
cacti.nj <- NJ(dist.dna(cacti.bin))
parsimony(tree=cacti.nj, data=aln.phy)
cacti.nj.optim <- optim.parsimony(cacti.nj, aln.phy)
cacti.nj.pratchet <- pratchet(aln.phy)
```

Cleaning up:
```{r}
unlink("aln.fasta")
unlink("seq.metadata.csv")
unlink("cacti.njtree.nwk")
```

