---
title: "Annotation of IQTree exports for CVX genes and full genome tree"
author: "A.T."
date: "5/20/2021"
output: html_document
---
#Notes: This is a long file and needs to be deconstructed to smaller loadable files if it is used again. treetemp, used later in the file, is an S4 object and its slots can be accessed by "treetemp@example". 

> This is a one-time package installation to help get libraries likely to be missing at the start.

```{r eval=FALSE}
knitr::opts_chunk$set(error = TRUE)
#install needed packages
install.packages('BiocManager')# This option gave B errors; repos=c('https://cran.rstudio.com'), ask=F)
BiocManager::install('DECIPHER')
BiocManager::install("ggtree")
BiocManager::install("phytools")
tinytex::install_tinytex()
install.packages('BiocManager', repos=c('https://cran.rstudio.com'), ask=FALSE)
BiocManager::install('DECIPHER')
install.packages("kableExtra")
```

## Loading Libraries

```{r load-libraries, message=FALSE}
library(ggtree) # tree plotting
library(DECIPHER)
library(phangorn)
library(phytools)
library(ape)
library(knitr)
library(DECIPHER)
library(treeio)
library(kableExtra)

```

Load metadata file with correct names - check which columns will be used for names and hosts.
```{r}
#reloading metadata file including the names, useful for running during debugging
reload.metadata <- function(){
  md <<- read.csv('../data/cvx_hostdata.csv', stringsAsFactors = FALSE)
  print("Metadata file loaded")
}
reload.metadata()
#cvx.newick.loc  <- "../data/iqtree-output/genbank_kr_consensus_extraction_al_trimmed.fasta.newick"
#cvx.newick.tree <- read.newick(cvx.newick.loc)
#use plot() and then tips.store for identifying nodes and tip numbers if needed:
#tips.store <- identify(newrerooted, nodes = TRUE, tips = TRUE,labels = TRUE, quiet = FALSE)$tips
```

Store tips for each available tree. Use tips.store values for clades.
```{r eval=FALSE}
search.column <- gsub(" ", "_", md$Organism)
search.column <- gsub("__", "_", search.column)
#Note that this only needed to be run once and is now outdated.
#rename.tips <- function(cladename, tipnames){
  #tipnames <- gsub("_", " ", tipnames)
  #tipnames <- gsub(":", "", tipnames)
#  for(i in 1:length(tipnames)){
#  md$Clade[grep(tipnames[i],search.column)] <<- cladename
#  print(md$Clade[grep(tipnames[i],search.column)])
#  print(tipnames[i])
#  print(i)
#  print(grep(tipnames[i],search.column))
  
#}
#}
#CVX Var A
#tips.store <- c("SRR11190795_Plant_FR-3_NC_002815", "SRR11603183_GW-peel-2_NC_002815", "SRR11603184_GW-peel-1_NC_002815", "SRR11190801_Plant_DH-2_NC_002815", "SRR11190798_Plant_DH-3_NC_002815", "SRR11190799_Plant_BSJ-3_NC_002815", "SRR11190800_Plant_BSJ-2_NC_002815", "SRR11190791_Plant_BSJ-1_NC_002815", "SRR11603182_GW-peel-3_NC_002815", "SRR11190802_Plant_DH-1_NC_002815", "SRR11190797_Plant_FR-1_NC_002815", "SRR11190796_Plant_FR-2_NC_002815", "Schlumbergera_truncata_19JSF_sty_NC_002815", "Schlumbergera_truncata_15H06_cons", "Schlumbergera_truncata_15H04_cons", "Schlumbergera_truncata_15H03_cons", "SRR11603189_RR-peel-3_NC_002815", "SRR11603190_RR-peel-2_NC_002815", "SRR11603186_RR-pulp-3_NC_002815", "SRR11603191_RR-peel-1_NC_002815", "Cactus_virus_X_LC128411.1", "SRR11603187_RR-pulp-2_NC_002815", "SRR11190792_Plant_BR-3_NC_002815", "SRR11190793_Plant_BR-2_NC_002815", "Cactus_virus_X_AF308158.2", "Cactus_virus_X_KM365479.1")
#rename.tips("Cactus Virus X Variant A", tips.store)

#CVX Var B
#tips.store <- c("Cactus_virus_X_JF937699.1","Cactus_virus_X_KM288846.1", "Cactus_virus_X_KM288847.1", "Cactus_virus_X_SCM51431")
#rename.tips("Cactus Virus X Variant B", tips.store)
#ZyVX Variant B1
#tips.store <- c("Zygocactus_virus_X_JF930326.1", "SRR11603191_RR-peel-1_NC_006059", "SRR11603190_RR-peel-2_NC_006059", "SRR11603187_RR-pulp-2_NC_006059", "SRR11603189_RR-peel-3_NC_006059", "SRR11603186_RR-pulp-3_NC_006059", "Zygocactus_virus_X_Strain_B1", "KM288842", "KM288843", "KM288844", "SRR11190801_Plant_DH-2_NC_006059", "SRR11190798_Plant_DH-3_NC_006059", "SRR11190797_Plant_FR-1_NC_006059", "SRR11190802_Plant_DH-1_NC_006059", "SRR11190796_Plant_FR-2_NC_006059")
#rename.tips("Zygocactus Virus X Variant B1", tips.store)

#Pitaya Virus X
#tips.store <- c("Mytcor_virus_1", "SRR11190800_Plant_BSJ-2_NC_024458", "SRR11190791_Plant_BSJ-1_NC_024458", "SRR11190793_Plant_BR-2_NC_024458", "SRR11190797_Plant_FR-1_NC_024458", "SRR11190796_Plant_FR-2_NC_024458", "Pitaya_virus_X", "SRR11190795_Plant_FR-3_NC_024458", "SRR11603187_RR-pulp-2_NC_024458", "SRR11603189_RR-peel-3_NC_024458", "SRR11603190_RR-peel-2_NC_024458", "SRR11603186_RR-pulp-3_NC_024458", "SRR11603191_RR-peel-1_NC_024458", "SRR11190801_Plant_DH-2_NC_024458", "SRR11190798_Plant_DH-3_NC_024458", "SRR11603183_GW-peel-2_NC_024458", "SRR11603182_GW-peel-3_NC_024458", "SRR11603184_GW-peel-1_NC_024458", "SRR11190802_Plant_DH-1_NC_024458")
#rename.tips("Pitaya Virus X", tips.store)

#Schlumbergera Virus X
#tips.store <- c("Schlumbergera_virus_X_KU854929", "Schlumbergera_virus_X_KP090203", "Schlumbergera_virus_X_Strain_K11", "SRR11190802_Plant_DH-1_NC_011659", "SRR11603183_GW-peel-2_NC_011659", "SRR11603189_RR-peel-3_NC_011659", "SRR11603187_RR-pulp-2_NC_011659", "SRR11603184_GW-peel-1_NC_011659", "SRR11190798_Plant_DH-3_NC_011659", "SRR11190801_Plant_DH-2_NC_011659")
#rename.tips("Schlumbergera Virus X", tips.store)

#Opuntia Virus X
#tips.store <- c("Opuntia_virus_X_KU854931", "Opuntia_virus_X_KY348771", "Opuntia_virus_Strain_CC10")
#rename.tips("Opuntia Virus X", tips.store)

#write.csv(md, "test1.csv")
```

Load csv files for relevant genes.
```{r}
#grep(,rownames(rdrp))
#Given clade name, returns indices of associated pairwise distance values within matrix.
reload.metadata()
rdrp <- read.csv('/Users/alexa/Desktop/cactusvirusx/data/data_v3/percent_seq_id/1_RdRp_perseq.csv', stringsAsFactors = FALSE, row.names = 1)
cp <- read.csv('/Users/alexa/Desktop/cactusvirusx/data/data_v3/percent_seq_id/5_CP_perseq.csv', stringsAsFactors = FALSE, row.names = 1)
full <- read.csv('/Users/alexa/Desktop/cactusvirusx/data/data_v3/percent_seq_id/0_full-aln_perseq.csv', stringsAsFactors = FALSE, row.names = 1)
```

Select specific information by clade designation within metadata file.
```{r clade-search}
cladesearch <- function (cladename){
  metarowname <- c()
  #fetch organism names from metadata file matching Clade column with clade name
  metarowname <- md$Organism[grep(cladename, md$Clade)]
  #change metadata organism names to match csv format
  metarowname <- gsub(" ", "_", metarowname)
  metarowname <- gsub(":", "", metarowname)

  #Note: Schlumbergera samples from gene specific ORF fastas have been renamed, which has been reflected in the metadata csv file.
  

  #Fetch the associated csv row/column number for items within list of organism names
  fullnum <<- c()
  for(i in 1:length(metarowname)){
       fullnum[i] <<- grep(metarowname[i], rownames(full))
       #print(metarowname[i])
  }
if (anyNA(fullnum)){
  print("Warning! One or more full.csv values is NA. Please check naming in all files.")
}
fullcomb <- combn(sort(unique(fullnum)), 2)

cpnum <<- c()
  for(i in 1:length(metarowname)){
       cpnum[i] <<- grep(metarowname[i], rownames(cp))
       #print(metarowname[i])
       #print(cpnum[i])
  }
if (anyNA(cpnum)){
  print("Warning! One or more cp values is NA. Please check naming in all files.")
}
cpcomb <- combn(sort(unique(cpnum)), 2)

  #exceptions for missing rows/columns in rdrp file
searchrdrp <- c("KM288843","Cactus_virus_X_KM288847.1",
                        "KM288842","KM288844","KM288845", 
                        "Cactus_virus_X__KM288847.1")
if(any(metarowname %in%  searchrdrp)){
    print("Note: removing values because sequences did not cover the rdrp gene. Sequence removed:")
  print(metarowname[(metarowname %in%
                      searchrdrp)])
    metarowname <- metarowname[!(metarowname %in%
                      searchrdrp)]
}
rdrpnum <<- c()
  for(i in 1:length(metarowname)){
       rdrpnum[i] <<- grep(metarowname[i], rownames(rdrp))
       #print(metarowname[i])
       #print(rdrpnum[i])
  }
if (anyNA(rdrpnum)){
  print("Warning! One or more rdrp values is NA. Please check naming in all files.")
  print(rdrpnum)
}
rdrpcomb <- combn(sort(unique(rdrpnum)), 2)

#Next, use indices to grab maximum and average pairwise distances.
#full
  #populate vector...
numtemp <- vector()
for (i in 1:length(fullcomb[1,])){
  numtemp[i] <- i
}
#use temporary vector to get values from full genome csv
for (i in 1:length(fullcomb[1,])){
  numtemp[i] <- full[fullcomb[1,i],fullcomb[2,i]]
}
#now calculate average and maximum values for these pairwise distances
fullmax <-c(max(numtemp))
fullavg <- c(mean(numtemp))
#repeat for the other genes
#rdrp
numtemp <- vector()
for (i in 1:length(rdrpcomb[1,])){
  numtemp[i] <- i
}
for (i in 1:length(rdrpcomb[1,])){
 numtemp[i] <- rdrp[rdrpcomb[1,i],rdrpcomb[2,i]]
}
rdrpmax <-c(max(numtemp))
rdrpavg <- c(mean(numtemp))
#cp
numtemp <- vector()
for (i in 1:length(cpcomb[1,])){
  numtemp[i] <- i
}
for (i in 1:length(cpcomb[1,])){
  numtemp[i] <- cp[cpcomb[1,i],cpcomb[2,i]]
}
cpmax <-c(max(numtemp))
cpavg <- c(mean(numtemp))
nums <- list("Max Values" = format(round(c(fullmax, rdrpmax, cpmax), 3), nsmall = 3), "Avg Values" = format(round(c(fullavg, rdrpavg, cpavg), 3), nsmall = 3) )
nums
}


```

Run cladesearch.
```{r clade-search-cva}
cva <- cladesearch("Cactus Virus X Variant A")
cvb <- cladesearch("Cactus Virus X Variant B")
zyvx <- cladesearch("Zygocactus Virus X Variant B1")
pvx <- cladesearch("Pitaya Virus X")
svx <- cladesearch("Schlumbergera Virus X")
opvx <- cladesearch("Opuntia Virus X")
#ZyVX Variant B1 #Pitaya Virus X, #Schlumbergera Virus X, #Opuntia Virus X

```

Fetch values for table.
```{r}
clades <- c("Cactus Virus X Variant A", " ", " ", "Cactus Virus X Variant B", " ", " ", "Zygocactus Virus X Variant B1", " ", " ", "Pitaya Virus X", " ", " ","Schlumbergera Virus X", " ", " ","Opuntia Virus X", " ", " ")
lnames <- rep(c("Full sequence", "RdRp", "Coat Protein"),6)
#matrix selection: matrix[rows, columns]
lmax <- c(cva$`Max Values`[1:3], cvb$`Max Values`[1:3], zyvx$`Max Values`[1:3] ,pvx$`Max Values`[1:3], svx$`Max Values`[1:3], opvx$`Max Values`[1:3])
lavg <- c(cva$`Avg Values`[1:3], cvb$`Avg Values`[1:3], zyvx$`Avg Values`[1:3], pvx$`Avg Values`[1:3], svx$`Avg Values`[1:3], opvx$`Avg Values`[1:3] ) 
```

Store values in table, then save table as .pdf file.

```{r}

sumstats <- cbind(clades, lnames, lmax, lavg)
kbl(sumstats, booktabs = T, format="html", col.names = c("Species", "Open Reading Frame (ORF)", "Maximum Pairwise Distance", "Average Pairwise Distance"))  %>% kable_styling() %>% save_kable("test.pdf")
```

##Trees:
Load trees for each gene.
```{r}
loc0 <- "../data/iqtree_0_full-aln_names/0_full-aln_names.fasta.treefile"
loc1 <- "/Users/alexa/Desktop/cactusvirusx/data/iqtree_1_RdRp_names/1_RdRp_names.fasta.treefile"
loc2 <- "/Users/alexa/Desktop/cactusvirusx/data/iqtree_2_TGB1_names/2_TGB1_names.fasta.treefile"
loc3 <- "/Users/alexa/Desktop/cactusvirusx/data/iqtree_3_TGB2_names/3_TGB2_names.fasta.treefile"
loc4 <- "/Users/alexa/Desktop/cactusvirusx/data/iqtree_4_TGB3_names/4_TGB3_names.fasta.treefile"
loc5 <- "/Users/alexa/Desktop/cactusvirusx/data/iqtree_5_CP_names/5_CP_names.fasta.treefile"
locs <- c(loc0, loc1, loc2, loc3, loc4, loc5)
names <- c("full", "rdrp", "tgb1", "tgb2", "tgb3", "cp")
```

Identify hosts for each tree and create phylogenies for each gene.
```{r}
gene.phylos <- function(namevector, locvector) {
  for (i in 1:length(namevector)){
  treetemp <- read.iqtree(locvector[i])
  treename <- namevector[i]
  treetempphylo <- as.phylo(treetemp)
#create vector for host information
  hostinfo <- c()
    hostinfo <- data.frame(label = gsub("_", " ", treetempphylo$tip.label), 
                           host=c(1:length(treetempphylo$tip.label)))
  search <- gsub(" ", "_", md$Organism)
  print(namevector[i])
#rename tree tips and populate vector
   for(j in 1:length(treetempphylo$tip.label)){
      hostinfo$host[j] <- md$host[grep(treetempphylo$tip.label[j],search)]
      treetempphylo$tip.label[j] <- gsub("_", " ", treetempphylo$tip.label[j])
      #uncomment these for troubleshooting:
      #print(treetempphylo$tip.label[j])
      #print(j)
      #print("Host:")
      #print(hostinfo$host[j])
      #print("Grep result")
      #print(md$host[grep(treetempphylo$tip.label[j],search)])
   }
#reroot tree and join host info, had trouble with the full_join function
treetemp <- root(treetempphylo, outgroup=c("Clover yellow mosaic virus"))
treetemp <- full_join(treetemp, hostinfo, by="label")

#prepare group for colors:
grp <- list(
  #CVX Var A
"Cactus Virus X Variant A" <- c("SRR11190795 Plant FR-3 NC 002815", "SRR11603183 GW-peel-2 NC 002815", "SRR11603184 GW-peel-1 NC 002815", "SRR11190801 Plant DH-2 NC 002815", "SRR11190798 Plant DH-3 NC 002815", "SRR11190799 Plant BSJ-3 NC 002815", "SRR11190800 Plant BSJ-2 NC 002815", "SRR11190791 Plant BSJ-1 NC 002815", "SRR11603182 GW-peel-3 NC 002815", "SRR11190802 Plant DH-1 NC 002815", "SRR11190797 Plant FR-1 NC 002815", "SRR11190796 Plant FR-2 NC 002815", "Schlumbergera truncata 19JSF sty NC 002815", "Schlumbergera truncata 15H06 cons", "Schlumbergera truncata 15H04 cons", "Schlumbergera truncata 15H03 cons", "SRR11603189 RR-peel-3 NC 002815", "SRR11603190 RR-peel-2 NC 002815", "SRR11603186 RR-pulp-3 NC 002815", "SRR11603191 RR-peel-1 NC 002815", "Cactus virus X LC128411.1", "SRR11603187 RR-pulp-2 NC 002815", "SRR11190792 Plant BR-3 NC 002815", "SRR11190793 Plant BR-2 NC 002815", "Cactus virus X AF308158.2", "Cactus virus X KM365479.1"),
#CVX Var B
"Cactus Virus X Variant B" <- c("Cactus virus X JF937699.1","Cactus virus X KM288846.1", "Cactus virus X KM288847.1", "Cactus virus X SCM51431"),
#ZyVX Variant B1
"Zygocactus Virus X Variant B1" <- c("Zygocactus virus X JF930326.1", "SRR11603191 RR-peel-1 NC 006059", "SRR11603190 RR-peel-2 NC 006059", "SRR11603187 RR-pulp-2 NC 006059", "SRR11603189 RR-peel-3 NC 006059", "SRR11603186 RR-pulp-3 NC 006059", "Zygocactus virus X Strain B1", "KM288842", "KM288843", "KM288844", "SRR11190801 Plant DH-2 NC 006059", "SRR11190798 Plant DH-3 NC 006059", "SRR11190797 Plant FR-1 NC 006059", "SRR11190802 Plant DH-1 NC 006059", "SRR11190796 Plant FR-2 NC 006059"),
"Zygocactus Virus X" <- c("KM288845"),
#Pitaya Virus X
"Pitaya Virus X" <- c("MG210801", "SRR11190800 Plant BSJ-2 NC 024458", "SRR11190791 Plant BSJ-1 NC 024458", "SRR11190793 Plant BR-2 NC 024458", "SRR11190797 Plant FR-1 NC 024458", "SRR11190796 Plant FR-2 NC 024458", "Pitaya virus X", "SRR11190795 Plant FR-3 NC 024458", "SRR11190799 Plant BSJ-3 NC 024458", "SRR11603187 RR-pulp-2 NC 024458", "SRR11603189 RR-peel-3 NC 024458", "SRR11603190 RR-peel-2 NC 024458", "SRR11603186 RR-pulp-3 NC 024458", "SRR11603191 RR-peel-1 NC 024458", "SRR11190801 Plant DH-2 NC 024458", "SRR11190798 Plant DH-3 NC 024458", "SRR11603183 GW-peel-2 NC 024458", "SRR11603182 GW-peel-3 NC 024458", "SRR11603184 GW-peel-1 NC 024458", "SRR11190802 Plant DH-1 NC 024458"),
#Schlumbergera Virus X
"Schlumbergera Virus X" <- c("Schlumbergera virus X KU854929", "Schlumbergera virus X KP090203", "Schlumbergera virus X Strain K11", "SRR11190802 Plant DH-1 NC 011659", "SRR11603183 GW-peel-2 NC 011659", "SRR11603189 RR-peel-3 NC 011659", "SRR11603187 RR-pulp-2 NC 011659", "SRR11603184 GW-peel-1 NC 011659", "SRR11190798 Plant DH-3 NC 011659", "SRR11190801 Plant DH-2 NC 011659"),
#Opuntia Virus X
"Opuntia Virus X" <- c("Opuntia virus X KU854931", "Opuntia virus X KY348771", "Opuntia virus Strain CC10")
)
#now visualize tree:
treetemp <- groupOTU(treetemp, grp)
par(mar=c(0, 0, 0, 0))
#L <- tree1name$Nnode + length(tree1name$tip.label)
ggtree(treetemp, ladderize=T, show.legend = FALSE)  +
  aes(color=group) +
#%<+% hostinfo+
  scale_color_manual(values=c("black", "#BE81EF", "#00A9E7", "#00B9A5", "#3FB13A", "#B39B00", "#E67E67", "#EC6DC4"))+
  #theme(legend.position="bottom")+
  #name tip labels:
  geom_tiplab(aes(), hjust=0, offset=0, align=T, linetype = "dotted")+
  #host tip labels:
  geom_tiplab(aes(label=host, subset = !is.na(host)), align = T, linetype = 'blank', na.rm=TRUE, offset=0.7) +
  #scale
  geom_treescale(x=0, y=60, fontsize = 3) +
  #node labels/bootstrap labels:
  geom_text2(aes(subset = !isTip & (node == 127)|(node == 124)|(node == 152)|(node == 153)|(node == 167)|(node == 102)|(node == 106)
                 , label=label), nudge_x = -0.03, nudge_y = 0.5)+
    xlim(0,4.2)
#Save to pdf format
filename = paste(namevector[i],"_bootstrap_tree.pdf",sep="")
ggsave(filename, width = 60, height = 40, units = "cm", limitsize = FALSE)
}
}
gene.phylos(names[1],locs[1])
gene.phylos(names, locs)
```

