---
title: "Annotation of IQTree exports for CVX genes and full genome tree"
author: "A.T."
date: "10/13/2021"
output: html_document
---
#Notes:  treetemp, used later in the file, is an S4 object and its slots can be accessed by "treetemp@example". 
> Packages

```{r eval=FALSE}
knitr::opts_chunk$set(error = TRUE)
#install needed packages
install.packages('BiocManager')# This option gave B errors; repos=c('https://cran.rstudio.com'), ask=F)
BiocManager::install('DECIPHER')
BiocManager::install("ggtree")
BiocManager::install("phytools")
tinytex::install_tinytex()
install.packages('BiocManager', repos=c('https://cran.rstudio.com'), ask=FALSE)
BiocManager::install('DECIPHER')
install.packages("kableExtra")
```
## Loading Libraries
```{r load-libraries, message=FALSE}
library(ggtree) # tree plotting
library(DECIPHER)
library(phangorn)
library(phytools)
library(ape)
library(knitr)
library(DECIPHER)
library(treeio)
library(kableExtra)
```
##Load metadata.
```{r}
reload.metadata <- function(){
  md <<- read.csv('../data/name_key.csv', stringsAsFactors = FALSE)
  print("Metadata file loaded")
}
reload.metadata()
```

##Trees:
Load trees for each gene.
```{r}
loc0 <- "../data/iqtree_0_full-aln_names/0_full-aln_names.fasta.treefile"
loc1 <- "../data/iqtree_1_RdRp_names/1_RdRp_names.fasta.treefile"
loc2 <- "../data/iqtree_2_TGB1_names/2_TGB1_names.fasta.treefile"
loc3 <- "../data/iqtree_3_TGB2_names/3_TGB2_names.fasta.treefile"
loc4 <- "../data/iqtree_4_TGB3_names/4_TGB3_names.fasta.treefile"
loc5 <- "../data/iqtree_5_CP_names/5_CP_names.fasta.treefile"
locs <- c(loc0, loc1, loc2, loc3, loc4, loc5)
names <- c("full", "rdrp", "tgb1", "tgb2", "tgb3", "cp")
```

Create phylogenies for each gene.
```{r}
gene.phylos <- function(namevector, locvector) {
  for (i in 1:length(namevector)){
  treetemp <- read.iqtree(locvector[i])
  treename <- namevector[i]
  treetempphylo <- as.phylo(treetemp)
#create vector for host information
  hostinfo <- c()
    hostinfo <- data.frame(label = treetempphylo$tip.label, 
                           host=c(1:length(treetempphylo$tip.label)))
  print(namevector[i])
#rename tree tips and populate vector
   for(j in 1:length(treetempphylo$tip.label)){
      hostinfo$host[j] <- md$hostgenus[grep(treetempphylo$tip.label[j],md$Name_updated)]
      
      #uncomment these for troubleshooting:
      #print(j)
      #print(treetempphylo$tip.label[j])
      
      #print("Host:")
      #print(hostinfo$host[j])
      #print("Grep result")
      #print(md$host[grep(treetempphylo$tip.label[j],search)])
   }
#reroot tree and join host info, had trouble with the full_join function
treetemp <- root(treetempphylo, outgroup=c("D29630.1"))
treetemp <- full_join(treetemp, hostinfo, by="label")

#prepare group for colors:
#grp <- list(
#Opuntia Virus X
#"Opuntia Virus X" <- c("Opuntia virus X KU854931", "Opuntia virus X KY348771", #"Opuntia virus Strain CC10")
#)
#now visualize tree:
#treetemp <- groupOTU(treetemp, grp)
par(mar=c(0, 0, 0, 2))
#L <- tree1name$Nnode + length(tree1name$tip.label)
ggtree(treetemp, ladderize=T, show.legend = FALSE)  +
  #aes(color=group) +
  #%<+% hostinfo+
  #scale_color_manual(values=c("black", "#BE81EF", "#00A9E7", "#00B9A5", "#3FB13A", "#B39B00", "#E67E67", "#EC6DC4"))+
  #theme(legend.position="bottom")+
  #name tip labels:
  geom_tiplab(aes(), hjust=0, offset=0, align=T, linetype = "dotted")+
  #host tip labels:
  geom_tiplab(aes(label=host, subset = !is.na(host)), align = T, linetype = 'dotted', na.rm=F, offset=0.4) +
  #scale
  geom_treescale(x=0, y=60, fontsize = 10) #+
  #node labels/bootstrap labels:
  #geom_text2(aes(subset = !isTip #& (node == 127)|(node == 124)|(node == 152)|(node == 153)|(node == 167)|(node == 102)|(node == 106)
    #             , label=label), nudge_x = -0.03, nudge_y = 0.5)+
    #xlim(0,1)
#Save to pdf format
filename = paste(namevector[i],"_bootstrap_tree.pdf",sep="")
ggsave(filename, width = 30, height = 30, units = "cm")
}
}
#gene.phylos(names[1],locs[1])
gene.phylos(names, locs)
```