---
title: "Annotation of IQTree exports for CVX genes and full genome tree"
author: "A.T."
date: "10/13/2021"
output: html_document
---
#Notes:  treetemp, used later in the file, is an S4 object and its slots can be accessed by "treetemp@example". 
## Loading Libraries
```{r load-libraries, message=FALSE}
library(ggtree) # tree plotting
library(DECIPHER)
library(phangorn)
library(phytools)
library(ape)
library(knitr)
library(treeio)
library(kableExtra)
library(dplyr)
```
##Load metadata.
```{r}
reload.metadata <- function(){
  md <<- read.csv('../../data/name_key.csv', stringsAsFactors = FALSE)
  print("Metadata file loaded")
}
reload.metadata()
```

##Trees:
Load trees for each gene.
```{r}
loc0 <- "../../data/iqtree_0_full-aln_names/0_full-aln_names.fasta.treefile"
loc1 <- "../../data/iqtree_1_RdRp_names/1_RdRp_names.fasta.treefile"
loc2 <- "../../data/iqtree_2_TGB1_names/2_TGB1_names.fasta.treefile"
loc3 <- "../../data/iqtree_3_TGB2_names/3_TGB2_names.fasta.treefile"
loc4 <- "../../data/iqtree_4_TGB3_names/4_TGB3_names.fasta.treefile"
loc5 <- "../../data/iqtree_5_CP_names/5_CP_names.fasta.treefile"
locs <- c(loc0, loc1, loc2, loc3, loc4, loc5)
names <- c("full", "rdrp", "tgb1", "tgb2", "tgb3", "cp")
```

Create phylogenies for each gene.
```{r}
gene.phylos <- function(namevector, locvector) {
  reload.metadata()
  for (i in 1:length(namevector)){
  treetemp <- read.iqtree(locvector[i])
  treename <- namevector[i]
#  treetempphylo <- as.phylo(treetemp)

  print(namevector[i])
#join tree and metadata
treetemp <- as.treedata(left_join(as_tibble(treetemp), md, by=c("label" = "Name_updated")))
is.rooted(treetemp)

tree.phy.temp <- root(treetemp@phylo, outgroup=c("D29630.1"))

ggtree(treetemp, ladderize=T, show.legend = FALSE, aes(color=treetemp@data$group))  +
  #aes(color=group) +
  #%<+% hostinfo+
  #scale_color_manual(name="Group",
 #                    values=c("#BE81EF", "#00A9E7", "#00B9A5", "#3FB13A", "#B39B00", #"#E67E67", "#EC6DC4", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#a6cee3","#1f78b4","#b2df8a","#33a02c"))+
  #host tip labels:
#  geom_tiplab(aes(label=host, subset = !is.na(host)), family='Helvetica',fontface= "bold", linetype="dotted", align=T, fontsize=3, na.rm=F, offset=1.5) +
    #newness tip labels:
  geom_tiplab(aes(label=new),size=10, color="black", family='Helvetica', align=T, na.rm=F, offset=1.4,vjust=0.8, linetype="blank", fontface=1) +
    #virus name tip labels:
  geom_tiplab(aes(), geom="label",fontsize=3, fill="white", offset.text=0, fontface= "bold",family='Helvetica', align=F, label.size = 0)+
  #scale
  geom_treescale(x=0.25, y=60, width=0.25, fontsize=4, linesize=1, offset=1,
                 color='black', family=font.default, label='Substitutions per site',
                 offset.label=-1)+
  ggplot2::xlim(0, 5.5)
  #theme(text=element_text(family="Helvetica"), plot.margin = unit(c(8,8,8,8), "mm")) 
  #node labels/bootstrap labels:
  #geom_text2(aes(subset = !isTip #& (node == 127)|(node == 124)|(node == 152)|(node == 153)|(node == 167)|(node == 102)|(node == 106)
    #             , label=label), nudge_x = -0.03, nudge_y = 0.5)+
    #xlim(NA,3)
#Save to pdf format
filename = paste(namevector[i],"_tr.pdf",sep="")
ggsave(filename, width = 30, height = 50, units = "cm")
}
}
#gene.phylos(names[1],locs[1])
gene.phylos(names, locs)
```