---
title: "Annotation of IQTree exports for CVX genes and full genome tree"
author: "A.T."
date: "10/13/2021"
output: html_document
---
#Notes:  treetemp, used later in the file, is an S4 object and its slots can be accessed by "treetemp@example". 
> Packages

```{r eval=FALSE}
knitr::opts_chunk$set(error = TRUE)
#install needed packages
install.packages('BiocManager')# This option gave B errors; repos=c('https://cran.rstudio.com'), ask=F)
BiocManager::install('DECIPHER')
BiocManager::install("ggtree")
BiocManager::install("phytools")
BiocManager::install("ape")
BiocManager::install("phangorn")
```
## Loading Libraries
```{r load-libraries, message=FALSE}
library(ggtree) # tree plotting

library(DECIPHER)
library(phytools)
library(ape)
library(treeio)
library(dplyr)
library(tidytree)
```
##Load metadata.
```{r}
reload.metadata <- function(){
  md <<- read.csv('../../data/name_key.csv', stringsAsFactors = FALSE)
  print("Metadata file loaded")
}
reload.metadata()
getwd()
```

##Trees:
Load trees for each gene.
```{r}
loc0 <- "../../data/iqtree_0_full-aln_names/0_full-aln_names.fasta.treefile"
loc1 <- "../../data/iqtree_1_RdRp_names/1_RdRp_names.fasta.treefile"
loc2 <- "../../data/iqtree_2_TGB1_names/2_TGB1_names.fasta.treefile"
loc3 <- "../../data/iqtree_3_TGB2_names/3_TGB2_names.fasta.treefile"
loc4 <- "../../data/iqtree_4_TGB3_names/4_TGB3_names.fasta.treefile"
loc5 <- "../../data/iqtree_5_CP_names/5_CP_names.fasta.treefile"
locs <- c(loc0, loc1, loc2, loc3, loc4, loc5)
names <- c("full", "rdrp", "tgb1", "tgb2", "tgb3", "cp")
treetemp <- read.iqtree(locs[1])
```

Create phylogenies for each gene.
```{r}
gene.phylos <- function(namevector, locvector) {
  reload.metadata()
  for (i in 1:length(namevector)){
  treetemp <- read.iqtree(locvector[i])
  treename <- namevector[i]
#  treetempphylo <- as.phylo(treetemp)

  print(namevector[i])
#join tree and metadata

treetemp@phylo <- ape::root(treetemp@phylo, outgroup=c("D29630.1"), resolve.root=TRUE)
treetemp
treetempjoined <- as.treedata(dplyr::left_join(as_tibble(treetemp), as_tibble(md), by=c("label"= "Name_updated")))
treetempjoined@phylo <- treetemp@phylo
par(mar = c(0, 0, 0, 0))
ggtree(treetempjoined, ladderize=T, aes(color=formal_tax))+
    #host tip labels:
   geom_tiplab(aes(label=host, subset = !is.na(host)), linetype="dotted", align=T, fontsize=3, na.rm=F, offset=0.8)+
    #virus name tip labels:
    geom_tiplab(aes(), geom="label",fontsize=3, fill="white", offset=0.1,family='Helvetica', align=F, label.size = 0)+

  #scale
    geom_treescale(x=0.25, y=60, width=0.25, fontsize=5, linesize=1, offset=1, color='black', label='Substitutions per site', offset.label=-1)+
   #node labels/bootstrap labels:
    geom_text2(aes(subset = !isTip , label=label), nudge_x = 0.02, size=3)+
    xlim(0,4.5)+
  theme(legend.position = "left")+
      #group labels
    scale_color_manual(name="Group", values=c("black", 
                                              "#0072B2",
                                              "#3FB13A", 
                                              "#BE81EF",
                                              "#B39B00", 
                                              "#E67E67", 
                                              "#EC6DC4", 
                                              "grey"))
#Save to pdf format
filename = paste(namevector[i],"_tr_formal_tax.pdf",sep="")
ggsave(filename, width = 70, height = 50, units = "cm")
}
}
gene.phylos(names,locs)
```